src/lib/routing-tables.js

```diff
@@ -99,16 +115,54 @@ class RoutingTables {
   _removeRoute (ledgerB, ledgerC, connectorFromBToC) {
     this.eachSource((tableFromA, ledgerA) => {
       if (ledgerA !== ledgerB) return
-      tableFromA.removeRoute(ledgerC, connectorFromBToC)
+      return tableFromA.removeRoute(ledgerC, connectorFromBToC)
     })
   }
 
   removeExpiredRoutes () {
+    let lostLedgerLinks = []
     this.eachRoute((routeFromAToB, ledgerA, ledgerB, nextHop) => {
-      if (routeFromAToB.isExpired()) {
-        this._removeRoute(ledgerA, ledgerB, nextHop)
+      if (!routeFromAToB.isLocal && routeFromAToB.isExpired()) {
```

Comment:
Why specify `!isLocal`? I know we don't want to expire local routes, but isn't that accomplished by not giving them an `expiresAt` when they are initialized?